src/modules/apps/update.ts

```diff
@@ -1,67 +1,90 @@
+import { Housekeeper } from '@vtex/api'
 import * as Bluebird from 'bluebird'
-import chalk from 'chalk'
 import * as ora from 'ora'
-import { isEmpty, map, pipe, prop, reject } from 'ramda'
+import { compose, concat, difference, filter, identity, map, path as ramdaPath, pluck, prop, union  } from 'ramda'
 
-import { apps } from '../../clients'
-import { parseLocator, toAppLocator, toMajorRange } from '../../locator'
+import { toMajorRange } from '../../locator'
 import log from '../../logger'
-import { createTable } from '../../table'
-import { diffVersions } from '../infra/utils'
 import { promptConfirm } from '../prompts'
-import { prepareInstall } from './install'
-import { appLatestVersion, isLinked } from './utils'
+import { matchedDepsDiffTable } from '../utils'
+import { getIOContext, IOClientOptions } from '../utils'
 
-const { listApps } = apps
 
 const promptUpdate = (): Bluebird<boolean> =>
   Promise.resolve(
     promptConfirm('Apply version updates?')
   )
 
-const sameVersion = ({ version, latest }: Manifest) => version === latest
+const sourceFilter = (source: string) => filter((obj: { source: string }) => includes(prop('source', obj), [source]))
 
-const extractAppLocator = pipe(prop('app'), parseLocator)
+const includes = (k: string, list: string[]) => list.indexOf(k) >= 0
 
-const updateVersion = (app) => {
-  app.version = app.latest
-  return app
-}
+const printAppsDiff = (resolvedUpdates: any, message: string, path: string[], filterFunction?: (obj: any) => boolean, pluckFunction?: (obj: any) => any) => {
```

Comment:
I think this method can be more simple.
It can receive only the `resolvedUpdates`, `message` and `type` (this one can be "apps" or  "infra"). Them this own method defines which is the filterFunction considering the value of `type`.